ec2/spark_ec2.py

```diff
@@ -346,9 +347,15 @@ def get_spark_ami(opts):
 # Returns a tuple of EC2 reservation objects for the master and slaves
 # Fails if there already instances running in the cluster's groups.
 def launch_cluster(conn, opts, cluster_name):
-    if opts.identity_file is None:
+    if opts.identity_file is None or \
+       not os.path.exists(opts.identity_file):
         print >> stderr, "ERROR: Must provide an identity file (-i) for ssh connections."
```

Comment:
It might be confusing to show the same error message in the case where an identity file is provided but cannot be found. Maybe show a separate error for that?

Also, what happens if you don't check `os.path.exists(opts.identity_file)` upfront and just let the launch proceed?